### PR TITLE
[AT-3970] enable pylint for utils and macros

### PR DESCRIPTION
**Pull request by seedlit**
Jira ticket [here.](https://up42.atlassian.net/browse/AT-3970)
Scope: enable pylint for all macros and utils files

Items:
* [ ] Ran test & live-tests
* [ ] Implemented (new) unit tests
* [ ] Removed credentials
* [ ] Removed argument pointing to staging
* [ ] Updated [documentation](sdk.up42.com)

For release:
* [ ] Bumped version
* [ ] Added changelog
* [ ] Updated announcement banner

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -30,8 +30,16 @@ You can check your current version with the following command:
 
 For more information, see [UP42 Python package description](https://pypi.org/project/up42-py/).
 
-## 0.37.0a4
+## 0.37.0a7
```
Comment (dustgalactic) [inline CHANGELOG.md:33]: This will be moved to another file as per this PR: https://github.com/up42/up42-py/pull/554

File: CHANGELOG.md [hunk 2]
```diff
@@ -30,8 +30,16 @@ You can check your current version with the following command:
...
+
+**March 07, 2024**
+
+- Adjusted `macros.py`, `utils.py`, and `test_utils.py` in accordance with Pylint checks.
 
+## 0.37.0a2
```
Comment (dustgalactic) [inline CHANGELOG.md:39]: Is it an intended change?

File: up42/utils.py [hunk 3]
```diff
@@ -113,9 +113,9 @@ def download_from_gcs_unpack(
     out_temp = tempfile.mkstemp()[1]
     with open(out_temp, "wb") as dst:
         try:
-            r = requests.get(download_url, stream=True)
+            r = requests.get(download_url, stream=True, timeout=TIMEOUT)
             r.raise_for_status()
-            for chunk in tqdm(r.iter_content(chunk_size=1024)):
+            for chunk in tqdm.tqdm(r.iter_content(chunk_size=1024)):
```
Comment (javidq) [inline up42/utils.py:118]: Let's move 1024 to CHUNK_SIZE constant

File: up42/utils.py [hunk 4]
```diff
@@ -165,25 +166,25 @@ def download_gcs_not_unpack(download_url: str, output_directory: Union[str, Path
...
                     dst.write(chunk)
         except requests.exceptions.HTTPError as err:
-            logger.debug(f"Connection error, please try again! {err}")
+            logger.debug("Connection error, please try again! %s", err)
             raise requests.exceptions.HTTPError(f"Connection error, please try again! {err}")
 
-    logger.info(f"Successfully downloaded the file at {out_fp}")
+    logger.info("Successfully downloaded the file at %s", out_fp)
     out_filepaths = [str(out_fp)]
     return out_filepaths
```
Comment (javidq) [inline up42/utils.py:184]: Can be inlined

File: up42/utils.py [hunk 5]
```diff
@@ -270,22 +275,22 @@ def any_vector_to_fc(
...
     geometry_error = "UP42 only accepts single geometries, the provided geometry {}."
     if len(fc["features"]) != 1:
-        logger.info(geometry_error.format("contains multiple geometries"))
+        logger.info("%s contains multiple geometries", geometry_error)
         raise ValueError(geometry_error.format("contains multiple geometries"))
 
     fc_type = fc["features"][0]["geometry"]["type"]
     if fc_type != "Polygon":
-        logger.info(geometry_error.format(f"is a {fc_type}"))
+        logger.info("%s is a %s", geometry_error, fc_type)
```
Comment (javidq) [inline up42/utils.py:289]: logger.error

File: up42/utils.py [hunk 6]
```diff
@@ -270,22 +275,22 @@ def any_vector_to_fc(
...
 
-def validate_fc_up42_requirements(fc: Union[dict, FeatureCollection]):
+def validate_fc_up42_requirements(fc: Union[dict, geojson.FeatureCollection]):
     """
     Validate the feature collection if it fits UP42 geometry requirements.
     """
     geometry_error = "UP42 only accepts single geometries, the provided geometry {}."
     if len(fc["features"]) != 1:
-        logger.info(geometry_error.format("contains multiple geometries"))
+        logger.info("%s contains multiple geometries", geometry_error)
```
Comment (javidq) [inline up42/utils.py:284]: logger.error

File: up42/utils.py (hunk 5 above)
Comment (javidq) [reply up42/utils.py:289]: Why do we even log here and above if we raise an error with the same message?

File: up42/utils.py [hunk 7]
```diff
@@ -338,7 +343,7 @@ def filter_jobs_on_mode(jobs_json: List[dict], test_jobs: bool = True, real_jobs
     if not selected_modes:
         raise ValueError("At least one of test_jobs and real_jobs must be True.")
     jobs_json = [job for job in jobs_json if job["mode"] in selected_modes]
-    logger.info(f"Returning {selected_modes} jobs.")
+    logger.info("Returning %s jobs.", selected_modes)
```
Comment (javidq) [inline up42/utils.py:346]: This can be moved to L345 and the jobs_json can be inlined

**Review by javidq — CHANGES_REQUESTED** [review]

File: up42/utils.py (hunk 5 above)
Comment (seedlit) [reply up42/utils.py:289]: valid point.
I will remove the log.

File: up42/utils.py (hunk 7 above)
Comment (seedlit) [reply up42/utils.py:346]: I didn't get `jobs_json can be inlined` fully.
Please see if the changes represent what you expected.

File: up42/utils.py [hunk 8]
```diff
@@ -165,25 +167,25 @@ def download_gcs_not_unpack(download_url: str, output_directory: Union[str, Path
...
-            logger.debug(f"Connection error, please try again! {err}")
+            logger.debug("Connection error, please try again! %s", err)
             raise requests.exceptions.HTTPError(f"Connection error, please try again! {err}")
 
-    logger.info(f"Successfully downloaded the file at {out_fp}")
-    out_filepaths = [str(out_fp)]
-    return out_filepaths
+        logger.info("Successfully downloaded the file at %s", out_fp)
+        out_filepaths = [str(out_fp)]
+        return out_filepaths
```
Comment (javidq) [inline up42/utils.py:185]: return [str(out_fp)]

**Review by javidq — CHANGES_REQUESTED** [review]

**Review by javidq — APPROVED** [review]

**Pull request merged**